### PR TITLE
Specs

### PR DESCRIPTION
**Pull request by vigehi**
Hello code reviewer

According to the project requirement we have worked on 

Add unit tests for all the methods in all the classes:

    Book
    Person
    Student
    Teacher
    Classroom
    Rental
    all classes that you created while implementing the Decorator design pattern

Thank you for taking your time to review our code

We highly appreciate your response

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      expect(person.name).to eq 'Jhon'
+      expect(person.parent_permission).to eq true
+    end
+  end
+
+  context 'should create an instance of Person class' do
+    peter = Person.new(13, 'Peter', false)
+    it 'Correct name' do
+      expect(peter.correct_name).to eq 'Peter'
+    end
```
Comment (Munsa1) [inline spec/person_spec.rb:21]: - [x] Kindly add a test which validates the name of a person for the `correct_name` method in `person.rb`

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @vigehi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly use the new `Readme` file template provided [here](https://github.com/microverseinc/readme-template/blob/master/README.md) as this is a new requirement.

Your Current Readme File
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb (hunk 1 above)
Comment (Whoistolu) [reply spec/person_spec.rb:21]: This test has already been written, hence, the project is complete.

**Review by Whoistolu — APPROVED** [review]
### Hi Team,
@vigehi and @tamana-Balkhi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**